docs/components/best-practices/security/csrf-protection.md

```diff
@@ -0,0 +1,115 @@
+
+CSRF protection is enabled by default for security. You can configure it using:
+
+**Application property:**  
+`camunda.security.csrf.enabled: true  # Set to 'false' to disable CSRF protection`
+
+**Environment variable:**  
+`CAMUNDA_SECURITY_CSRF_ENABLED=true  # Set to false to disable CSRF protection`
+
+:::caution
+Disabling CSRF protection is not recommended for production environments as it leaves your application vulnerable to cross-site request forgery attacks.
+:::
+
+## Authentication method differences
+
+### Basic authentication
+
+- CSRF tokens are generated upon successful login via the `/login` endpoint.
+- The token is included in the response header and stored in a secure cookie.
+- Subsequent requests must include the token in the `X-CSRF-TOKEN` header.
+- Tokens are cleared on logout.
+
+### OIDC (OpenID Connect) authentication
+
+- CSRF protection works similarly to Basic Auth.
+- Tokens are generated after successful OAuth2/OIDC authentication.
+- The same `X-CSRF-TOKEN` header mechanism applies.
+- Compatible with both JWT bearer tokens and session-based authentication.
+
+## Protected vs unprotected paths
+
+### Protected paths (require CSRF token)
+
+- `/api/**` – API endpoints (except specifically excluded paths)
+- `/v1/**`, `/v2/**` – Versioned API endpoints
+- All state-changing operations (POST, PUT, DELETE, PATCH)
+
+### Unprotected paths (no CSRF token required)
+
+- `/actuator/**` – Health and monitoring endpoints
+- `/swagger-ui/**` – API documentation
+- `/v2/license` – Public license endpoint
+- `/error` – Error handling
+- Authentication endpoints (`/login`, `/logout`)
+- Safe HTTP methods (GET, HEAD, OPTIONS, TRACE)
+
+## Implementation details
+
+### Request matching logic
+
+The implementation determines whether CSRF protection should be applied based on:
+
+- **HTTP method**: Safe methods (GET, HEAD, TRACE, OPTIONS) bypass CSRF checks.
+- **Request path**: Unprotected paths are excluded from CSRF validation.
+- **Session presence**: API calls with active browser sessions require CSRF tokens.
+- **Swagger UI**: Requests originating from Swagger UI are exempt.
+
+## Troubleshooting
+
+### Common issues
+
+**403 Forbidden on POST/PUT/DELETE requests**
+
+- Ensure the `X-CSRF-TOKEN` header is included.
+- Verify the token matches the one in the cookie.
+- Check that cookies are being sent with requests.
+
+**Token not received**
+
+- Make an authenticated GET request first to receive a token.
+- Ensure cookies are enabled in the client.
+- Check for CORS issues if making cross-origin requests.
+
+**Token mismatch errors**
+
+- Tokens may expire with the session.
+- Re-authenticate to get a new token.
+- Ensure you're using the most recent token.
+
+## Security considerations
+
+- Always use HTTPS in production to prevent token interception.
+- CSRF protection complements but doesn't replace proper authentication.
+- Consider additional security headers configured in the security settings.
```

Comment:
I suggest to only have this point if we can link to documentation on those settings.